src/lib.rs

```diff
@@ -135,7 +135,11 @@ r#"TypeLayoutInfo {
 #![cfg_attr(doc, feature(doc_auto_cfg))]
 // optional feature-gated features
 // https://github.com/rust-lang/rust/issues/94039
-#![cfg_attr(feature = "impl-atomics", feature(cfg_target_has_atomic))]
+#![feature(cfg_target_has_atomic)]
```

Comment:
```suggestion
#![cfg_attr(feature = "impl-atomics", feature(cfg_target_has_atomic))]
```

Reply:
Ah yes I did try this, as well as always enabling the `integer_atomics` feature when `impl-atomics` is enabled, however then I get 
```
  | error[E0635]: unknown feature `integer_atomics`
  |    --> const-type-layout\src\lib.rs:139:47
  |     |
  | 139 | #![cfg_attr(feature = "impl-atomics", feature(integer_atomics))]
```
from the ptx compilation. So perhaps the original commit is the only way to make this work.